db/helpers.go

```diff
@@ -129,6 +129,13 @@ func (ms *MongoStorage) createIndexes() error {
 	if _, err := ms.verifications.Indexes().CreateOne(ctx, verificationCodeIndex); err != nil {
 		return fmt.Errorf("failed to create index on code for verifications: %w", err)
 	}
+	// create an index for the 'type' field on user verifications
+	verificationTypeIndex := mongo.IndexModel{
```

Comment:
@lucasmenendez What does an index to a non unique field mean abstractly? How is this is going to optimize searching?

Reply:
fixed, ill remove this index and modify the index over `code` to be over the tuple `code-type` which is always unique.